lib/ws-kernel.coffee

```diff
@@ -0,0 +1,84 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+uuid = require 'uuid'
+services = require('./jupyter-js-services-shim')
+
+Kernel = require './kernel'
+
+module.exports =
+class WSKernel extends Kernel
+    constructor: (kernelSpec, @session) ->
+        super kernelSpec
+
+        @session.statusChanged.connect => @_onStatusChange()
+        @_onStatusChange() # Set initial status correctly
+
+    interrupt: ->
+        @session.kernel.interrupt()
+
+    restart: ->
+        @session.kernel.restart()
+
+    _onStatusChange: ->
+        @statusView.setStatus @session.status
+
+    _onIOPub: (message, onResults) ->
+        msg_type = message.header.msg_type
+
+        # TODO(nikita): implement support for watches
+
+        if onResults?
+            console.log('got msg')
+            console.log(message)
+            result = @_parseIOMessage(message)
+            if result?
+                onResults result
```

Comment:
Note that this is inside an `if onResults?` block. Is double-checking the accepted coding style here?